src/analysis/ml/ml.py

```diff
@@ -0,0 +1,137 @@
+import pandas as pd
+from typing import Iterable
+from pathlib import Path, PurePath
+import os
+
+import matplotlib.pyplot as plt
+plt.switch_backend('Agg')
+from sklearn.decomposition import PCA
+from sklearn.preprocessing import StandardScaler
+import seaborn as sns
+
+from sklearn.cluster import AgglomerativeClustering
+from scipy.cluster.hierarchy import dendrogram
+import numpy as np
+
+def summarize_networks(file_paths: Iterable[Path]) -> pd.DataFrame:
+
+    # creating a tuple that contains the algorithm column name and edge pairs
+    edge_tuples = []
+    
+    for file in file_paths:
+        
+        try:
+           
+           # collecting and sorting the edge pairs per algortihm
+            with open(file,'r') as f:
+                # under the assumption that it is single chars (maybe)
+                # split on space char 
+                # take the first and 0th elements and sort those 
+                lines = [line[:3] for line in f.readlines()]
+            
+            line = []
+            for char in lines:
+                newChar = char.replace(' ','')
+                line.append(newChar)
+                
+            # to deal with edges are the same but not ordered the same
+            e = [''.join(sorted(ele)) for ele in line]
+
+            
+            # getting the algorithm name
+            p = PurePath(file)
+            p.parts[-2]
```

Comment:
I'm not sure if it's happening here or elsewhere in the code, but in the hierarchical clustering figure I get one unlabeled algorithm.
![data0-hac](https://user-images.githubusercontent.com/3532898/221995083-b485a033-0d68-4300-9fb4-0c2c62ef84b8.png)

Reply:
I'm getting this too...

Reply:
I figured out what the number is. In total there are 10 algorithms being processed, and what's being shown is 6 of the algorithm's. I think the (4) is the other four algorithms that are "being shown". 
I have no idea why this is happening, maybe from the method I took from scikit learn, but at least it's not just some random number

Reply:
This was happening for dataset 1 and 0, and this theory seems to at least check out for those two

Reply:
I dug into this and ultimately the issue is due to the arguments we are providing the `dendrogram` function. I'll make a suggestion separately that fixes it and comment more there.

There was some helpful background in this old issue talking about adding dendrogram plotting support to sklearn `https://github.com/scikit-learn/scikit-learn/pull/3464`.